src/mapping-sensor.h

```diff
@@ -0,0 +1,46 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "core/extension.h"
+
+namespace librealsense {
+    class mapping_sensor : public recordable<mapping_sensor>
+    {
+    public:
+        virtual ~mapping_sensor() = default;
+
+        void create_snapshot(std::shared_ptr<mapping_sensor>& snapshot) const override;
```

Comment:
Please list in the ticket to verify record & playback (once we can..)

Reply:
Done - new ticket opened for this issue